rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,241 @@
+                    if new_best_match:
+                        # the new match is a closer match
+                        if new_best_match[0] != best_match:
+                            best_match = new_best_match[0]
+                            best_match_section = sec_name
+
+        return best_match_section or rebase_spec_file.get_main_files_section()
+
+    @classmethod
+    def _correct_missing_files(cls, rebase_spec_file, files):
+        """Adds files found in sources to the SPEC file."""
+        macros = [m for m in rebase_spec_file.macros if m['name'] in MacroHelper.MACROS_WHITELIST]
+        macros = MacroHelper.expand_macros(macros)
+        # ensure maximal greediness
+        macros.sort(key=lambda k: len(k['value']), reverse=True)
+
+        for file in files:
+            section = cls._get_best_matching_files_section(rebase_spec_file, file)
+            substituted_path = MacroHelper.substitute_path_with_macros(file, macros)
+            rebase_spec_file.spec_content.sections[section].insert(0, substituted_path)
+            logger.info('Added %s to %s section', substituted_path, section)
+
+    @classmethod
+    def _correct_deleted_files(cls, rebase_spec_file, files):
+        """Removes files missing in sources from the SPEC file."""
+        for sec_name, sec_content in six.iteritems(rebase_spec_file.spec_content.sections):
+            if sec_name.startswith('%files'):
+                i = 0
+                while i < len(sec_content):
+                    line = sec_content[i]
+                    if line != '':
+                        split_line = line.split(' ')
+                        directives = []
+                        for element in reversed(split_line):
+                            if element in cls.FILES_DIRECTIVES:
+                                if element == '%doc' or element == '%license':
+                                    insert_path = MacroHelper.expand(
+                                        '%{_defaultdocdir}' if element == '%doc' else '%{_defaultlicensedir}')
+                                    # expand the files on the line
+                                    for k in range(len(split_line) - 1):
+                                        element2 = line[k]
+                                        if not element2.startswith('%'):
+                                            split_line[k] = os.path.join(insert_path, element2)
+                                directives.insert(0, element)
+                                split_line.remove(element)
+
+                        expanded_line = [MacroHelper.expand(f) for f in split_line]
+
+                        j = 0
+                        while j < len(expanded_line):
+                            file = expanded_line[j]
+                            for deleted_file in reversed(files):
+                                if fnmatch.fnmatch(deleted_file, file):
```

Comment:
Same as above, you can test the negative case first, and increment the counter and continue with the loop, and then do the actual work, avoiding `else` altogether. It saves you one indentation level and it is more compact and readable in my opinion.